src/marqo/s2_inference/clip_utils.py

```diff
@@ -182,15 +186,30 @@ def download_image_from_url(image_path: str, media_download_headers: dict, timeo
     headers.update(media_download_headers)
     c.setopt(pycurl.HTTPHEADER, [f"{k}: {v}" for k, v in headers.items()])
 
+    # callback to check file size for video and audio
+    if modality in ["video", "audio"]:
+        max_size = EnvVars.MARQO_MAX_VIDEO_AUDIO_SEARCH_FILE_SIZE
+        def progress(download_total, downloaded, upload_total, uploaded):
+            if downloaded > max_size:
+                return 1  # Abort the download
```

Comment:
Can we raise ImageDownloadError directly here?

Reply:
done

Reply:
currently not, we will get a "Callback aborted" error

Reply:
So even if we raise an error here, it will caught and convertd to a `Callback aborted` error?

Reply:
yeah, I tried and it gave me a Callback aborted error when I raised here